src/search/merge_and_shrink/transition_system.h

```diff
@@ -94,51 +93,50 @@ class TransitionSystem {
     const int num_variables;
     std::vector<int> incorporated_variables;
 
+    const Labels &labels;
     /*
       All locally equivalent labels are grouped together, and their
-      transitions are only stored once for every such group, see below.
-
-      LabelEquivalenceRelation stores the equivalence relation over all
-      labels of the underlying factored transition system.
-    */
-    std::unique_ptr<LabelEquivalenceRelation> label_equivalence_relation;
-
-    /*
-      The transitions of a label group are indexed via its ID. The ID of a
-      group does not change, and hence its transitions are never moved.
-
-      We tested different alternatives to store the transitions, but they all
-      performed worse: storing a vector transitions in the label group increases
-      memory usage and runtime; storing the transitions more compactly and
-      incrementally increasing the size of transitions_of_groups whenever a
-      new label group is added also increases runtime. See also issue492 and
-      issue521.
+      transitions are only stored once for every such group. Each such group
+      is represented by a local label (LocalLabelInfo). Local labels can be
+      mapped back to the set of labels they represent. Their cost is
+      the minimum cost of all represented labels.
     */
-    std::vector<std::vector<Transition>> transitions_by_group_id;
+    std::vector<int> label_to_local_label;
+    std::vector<LocalLabelInfo> local_label_infos;
 
     int num_states;
     std::vector<bool> goal_states;
     int init_state;
 
     /*
-      Check if two or more labels are locally equivalent to each other, and
-      if so, update the label equivalence relation.
+      Check if two or more local labels are locally equivalent to each other,
+      and if so, merge them and store their transitions only once.
     */
     void compute_locally_equivalent_labels();
 
-    const std::vector<Transition> &get_transitions_for_group_id(int group_id) const {
-        return transitions_by_group_id[group_id];
-    }
-
     // Statistics and output
     int compute_total_transitions() const;
     std::string get_description() const;
+
+    // Consistency
```

Comment:
I'm not sure I like this line as a comment, especially since "consistency" is also a heuristic property, and also `dump_label_mapping` doesn't sound like something I'd search for under "Consistency".